.github/workflows/js-css-linting.yml

```diff
@@ -28,13 +40,21 @@ jobs:
 
         steps:
             - uses: actions/checkout@v3
-            - name: Setup node version and npm cache
+
+            - name: Cache node_modules
+              id: cache-node-modules
+              uses: actions/cache@v3
+              env:
+                  cache-name: cache-node-modules
+              with:
+                  path: node_modules
+                  key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+
+            - name: Setup node version
               uses: actions/setup-node@v3
               with:
                   node-version-file: '.nvmrc'
-                  cache: 'npm'
-            - name: Install Node dependencies
-              run: npm ci --no-optional
```

Comment:
We have removed the `Install Node dependencies` step from the `JSLintingCheck` job, I think we won't need the `Cache node_modules` & `Setup node version` steps in this case!

Reply:
This is a bit confusing the way the actions work, but we need both of those. The code is the same for add & restoring from the cache, so we need the `Cache node_modules` everywhere we rely on `node_modules` to be installed. In some jobs, they might be doing the caching, in others they will retrieve from the cache.

The `Setup node version` equally is needed in every job as we want to run all jobs on the same node version, and it doesn't persist between jobs.

Reply:
Thanks for the answer! 🙌